### DISCUSSION TITLE
Feature idea: job-level timeouts

### DISCUSSION THREAD

**Discussion by crowecawcaw** [OpenJobDescription/openjd-specifications#41] [Ideas]
Currently OJD supports timeouts per action which are useful for a job template author who knows how long individual actions should take. I suspect that end users who use a job template think in terms of how long the job - not the individual actions - should take. While submitting a render job, an artist might think, "This render finishes in an hour on my machine, so if it takes more than a couple hours of compute time on the farm, something's gone wrong."

What do you all think about supporting job level timeouts in addition to timeouts on individual actions?

I expect the design would take some thought since it's not obvious how to factor onEnter and onExit actions into the total job runtime (or other session actions for that matter). I expect the implementation would also be difficult; action timeouts can be enforced just by the worker running them but job timeouts would need tracking and enforcement by the service that's scheduling the tasks across workers.

**Comment by rickrams** [comment]
It's an interesting idea - "this render finishes in an hour on my machine" is going to translate most often to a sense of the duration of a task though - not a sense of the duration of 10 tasks, 100 tasks, etc? A job timeout could also optionally include scheduling time - i.e. if this job isn't picked up and worked on within XYZ then don't bother?
